contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1922,7 +1923,14 @@ pltsql_store_view_definition(const char *queryString, ObjectAddress address)
 	new_record[0] = Int16GetDatum(dbid);
 	new_record[1] = CStringGetTextDatum(logical_schemaname);
 	new_record[2] = CStringGetTextDatum(NameStr(form_reltup->relname));
-	new_record[3] = CStringGetTextDatum(queryString);
+	if (original_query)
+		new_record[3] = CStringGetTextDatum(original_query);
+	/*
+	 * sp_describe_first_result_set internally translates to a "CREATE VIEW" statement,
+	 * we don't want to save the view definition for that case.
+	 */
+	else
+		new_record[3] = CStringGetTextDatum("NULL");
```

Comment:
In case the original_query is NULL, we have inserted "queryString" in the catalog.
This executes for a scenario where we internally translate "sp_describe_first_result_set" to a "CREATE VIEW" statement, though we don't save anything in the catalog since it's not an actual view. 
IMO we can't have a test case for this.